docs/Configuration/yaml-files.md

```diff
@@ -81,6 +83,13 @@ policies:
   platform: darwin
   critical: false
   calendar_event_enabled: false
+- name: Firefox on Linux installed and up to date
+  platform: linux
+  description: "This policy checks that Firefox is installed and up to date."
+  resolution: "Install Firefox version 129.0.2 or higher."
+  query: "SELECT 1 FROM deb_packages WHERE name = 'firefox' AND version_compare(version, '129.0.2') >= 0;"
```

Comment:
Let's use 129.0.3 here; I know the query for 129.0.2 is bugged based on the rest of the build string Mozilla provides, so this is a bad example 😬

Reply:
I don't follow sorry, this example has been working for me on my Ubuntu VM.

Reply:
So, I tested an upgrade path from Firefox 129.0.0, as https://ftp.mozilla.org/pub/firefox/releases/129.0/linux-x86_64/en-US/firefox-129.0.deb. The version string on that .deb is `129.0~build2`, which appears to _pass_ the version_compare call against 129.0.2, so it wouldn't register as outdated and the upgrade wouldn't be pushed.

It looks like this isn't a one-off either, as e.g. the current 129.0.2 for amd64 is build1.

I was able to repro this in my QA of #20895 / #19551. Let me know if you need help repro'ing this.

Reply:
Oh gotcha. Makes sense.

Isn't this broken for 129.0.3?
```
osquery> select version_compare('129.0~build2', '129.0.3') >= 0;
+-------------------------------------------------+
| version_compare('129.0~build2', '129.0.3') >= 0 |
+-------------------------------------------------+
| 1                                               |
+-------------------------------------------------+
```

We could amend the query to remove any `build` suffixes or add a patch version `0` when it isn't present.

Reply:
Ah, yep.

And yeah, we should take that into account on the query if we're using FF as an example. I'm sure it's not the only package with an issue like this.

Reply:
Yes, very hard to detect all edge cases with all apps, but... that's the point of this feature, you write a query per app (trade off ease of use vs flexibility). For most cases the compare_version SQL will work, if it doesn't you can modify it for your custom app/scripts.